### ISSUE TITLE
Re-export HttpStatus from dart:html

### ISSUE THREAD

**Issue by natebosch** [dart-lang/sdk#34318]
Since we allow imports to `dart:io` from code running with DDC/Dart2JS as long as none of the runtime behavior is used (as a hack to get around some other limitations) some code is using constants from `dart:io` like the ones in `HttpStatus`. There seem to be legitimate reasons to use these constants on the web - can we export them from both SDK libraries?

cc @vsmenon

**Comment by jmesserly** [comment]
Also CC @terrylucas 

+1 though, seems like an easy fix.

**Comment by kevmoo** [comment]
I'd rather NOT add more things to dart: libraries.

We should be moving the other way, IMHO

**Comment by natebosch** [comment]
`HttpHeaders` also looks like it could be useful on the web.

**Comment by jmesserly** [comment]
> I'd rather NOT add more things to dart: libraries.

yeah I totally get where you're coming from (and agree). What's different about this one: we aren't really adding anything, just re-exporting something. Also integer constants like the ones on HttpStatus are "free" in dart2js (& should be in DDC too).

**Comment by grouma** [comment]
Can we make a decision on this one so I can correctly communicate to our internal users? I am nearing the completion of the 'strict platforms' work which is directly impacted by this. Users are not able to import form dart:io and dart:html in the same library. Users should know if they need to create their own static utility class or wait until html supports these constants.

**Comment by vsmenon** [comment]
Re-exporting in `dart:html` sgtm.  We may want to eventually move this to a private (`dart:_*`) library and have `dart:io` and `dart:html` re-export that.

@rakudrama - any concerns for dart2js / tree-shaking?  Right now, web folks are importing `dart:io` to get these.

**Comment by rakudrama** [comment]
If you are suggesting

html_dart2js.dart:
```dart
export 'dart:io' show HttpStatus;
```

then sgtm.

But since dart2js will need to do some more work to load dart:io even when it is not used, and you will have to change the dart:html templates and regenerate the library, it is not much more effort to create dart:_http_constants now rather than later.

**Comment by vsmenon** [comment]
@a-siva @zanderso - are you guys ok with moving some constants (e.g., `HttpStatus` ones) to a shared internal `dart:_*` file and re-exporting from `dart:io` and `dart:html`?

We're trying to snip web dependences on `dart:io`.

**Comment by zanderso** [comment]
I think that should be fine, yes.

**Comment by terrylucas** [comment]
dart:html changed with corresponding test CL pending.

**Comment by terrylucas** [comment]
Fixed with [CL](https://dart-review.googlesource.com/c/sdk/+/89340)
